compareBuckets/index.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable max-len */
+/* eslint-disable no-console */
+/* eslint-disable comma-dangle */
+
+const { Logger } = require('werelogs');
+
+const { compareBuckets } = require('./compareBuckets');
+
+const DEFAULT_WORKERS = 100;
+const DEFAULT_LOG_PROGRESS_INTERVAL = 10;
+const DEFAULT_LISTING_LIMIT = 1000;
+
+const {
+    DST_BUCKETD_HOSTPORT,
+    SRC_BUCKETD_HOSTPORT,
+    SRC_BUCKET,
+    DST_BUCKET,
+} = process.env;
+
+const WORKERS = (
+    process.env.WORKERS
+        && Number.parseInt(process.env.WORKERS, 10)) || DEFAULT_WORKERS;
+
+const LOG_PROGRESS_INTERVAL = (
+    process.env.LOG_PROGRESS_INTERVAL
+        && Number.parseInt(process.env.LOG_PROGRESS_INTERVAL, 10))
+      || DEFAULT_LOG_PROGRESS_INTERVAL;
+
+const SRC_KEY_MARKER = process.env.SRC_KEY_MARKER || '';
+const DST_KEY_MARKER = process.env.DST_KEY_MARKER || '';
+
+const LISTING_LIMIT = (
+    process.env.LISTING_LIMIT
+        && Number.parseInt(process.env.LISTING_LIMIT, 10))
+      || DEFAULT_LISTING_LIMIT;
+
+const USAGE = `
+compareBuckets.js
+
+This script compares the current objects in a bucket of two bucketd databases.
+
+Usage:
+    node compareBuckets.js
+
+Mandatory environment variables:
+    SRC_BUCKETD_HOSTPORT: ip:port of the source bucketd endpoint
+    DST_BUCKETD_HOSTPORT: ip:port of the destination bucketd endpoint
+    SRC_BUCKET: bucket to be scanned
+    DST_BUCKET: bucket to be scanned
+
+Optional environment variables:
+    SRC_KEY_MARKER: key to continue listing from on the source bucket
```

Comment:
I think the user should only pass a single option here to resume from a point, up to the script to set its state and resume the listing from the same point on both bucketds.